src/main/java/org/folio/rest/impl/PermissionUtils.java

```diff
@@ -1,11 +1,31 @@
 package org.folio.rest.impl;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 import org.folio.rest.jaxrs.model.OkapiPermission;
 import org.folio.rest.jaxrs.model.Permission;
+import org.folio.rest.jaxrs.model.PermissionNameListObject;
+import org.folio.rest.persist.PostgresClient;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
 
 public class PermissionUtils {
 
+  private static final String SELECT_DEPRECATED_PERMS = "select id::text as id, jsonb->>'permissionName' as name "
+      + "from %s_mod_permissions.permissions where jsonb->>'deprecated' = 'true'";
+  private static final String PURGE_DEPRECATED_PERMS = "delete from %s_mod_permissions.permissions "
+      + "where jsonb->>'deprecated' = 'true'";
+  private static final String PURGE_DEPRECATED_SUB_PERMS = "update %s_mod_permissions.permissions "
+      + "set jsonb = jsonb_set(jsonb, '{subPermissions}', (jsonb->'subPermissions')::jsonb - array['%s', '%s']) "
+      + "where jsonb->'subPermissions' ?| array['%s', '%s']";
+  private static final String PURGE_DEPRECATED_PERMS_USERS = "update %s_mod_permissions.permissions_users "
+      + "set jsonb = jsonb_set(jsonb, '{permissions}', (jsonb->'permissions')::jsonb - array['%s', '%s']) "
+      + "where jsonb->'permissions' ?| array['%s', '%s']";
+
   private PermissionUtils() {
```

Comment:
LGTM, but not an expert here.

Reply:
I am no expert either. I just did what the pg doc says. :)